pkg/cmd/pulumi/install/install.go

```diff
@@ -89,10 +90,42 @@ func NewInstallCmd(ws pkgWorkspace.Context) *cobra.Command {
 				}
 				pluginPath, err := workspace.DetectPluginPathFrom(cwd)
 				if err == nil && pluginPath != "" {
+					// We're in a plugin. First install the packages specified
+					// in the plugin project file, and then install the plugin's
+					// dependencies.
+
 					proj, err := workspace.LoadPluginProject(pluginPath)
 					if err != nil {
 						return err
 					}
+
+					pctx, err := plugin.NewContextWithRoot(ctx,
+						cmdutil.Diag(),
+						cmdutil.Diag(),
+						nil, // host
+						cwd, // pwd
+						cwd, // rot
+						proj.Runtime.Options(),
+						false, // disableProviderPreview
+						nil,   // tracingSpan
+						nil,   // Plugins
+						proj.GetPackageSpecs(),
+						nil, // config
+						nil, // debugging
+					)
+					if err != nil {
+						return err
+					}
+
+					// Cloud registry is linked to a backend, but we don't have
+					// one available in a plugin. Use the unauthenticated
+					// registry.
+					reg := unauthenticatedregistry.New(cmdutil.Diag(), env.Global())
```

Comment:
So this is going to fail if anyone has dependencies that depend on the cloud registry?  Probably okay as a first step, but might be worth a `TODO` and an issue to remind us?

Looking below, this is because an authenticated registry is tied to a project. Curious if it shouldn't be tied to a logged in user :thinking: